### DISCUSSION TITLE
Welcome to homeassistant-addon-victoriametrics discussions!

### DISCUSSION THREAD

**Discussion by fuslwusl** [VictoriaMetrics-Community/homeassistant-addon-victoriametrics#1] [Announcements]
## 👋 Welcome!
Here is the place to give feedback to the Home Assistant VictoriaMetrics Add-on.

  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by paulevw** [comment]
Hello,
I'm struggling just with the installation ^^

System: PI4, Docker, Supervised

Trying to install the addon in the addon-store > repository
https://github.com/fuslwusl/homeassistant-addon-victoriametrics

with the result:
Cmd('git') failed due to: exit code(128) cmdline: git clone -v --recursive --depth=1 --shallow-submodules https://github.com/fuslwusl/homeassistant-addon-victoriametrics/releases/tag/1.8.10 /data/addons/git/210b5183 stderr: 'Cloning into '/data/addons/git/210b5183'... fatal: https://github.com/fuslwusl/homeassistant-addon-victoriametrics/releases/tag/1.8.10/info/refs not valid: is this a git repository? '

???
THX

**Reply by fuslwusl** [reply]
You have to checkout the addon and place the folder victoria-metrics in the addons/ folder of your homeassistant installation.
/addons/victoria-metrics
/shared/...
/config/...

**Reply by paulevw** [reply]
THX for the quick reply,
Well, i can get to the addons folder via ssh (in Home assistant), but **how to get the files there ?**, 
wget seems not to work (or i'm no able to use it correctly)
samba currently does not work for me 
WinSCP does not get me to the HA-Docer container ...
Sorry, not the most experienced user ^^^

**Reply by fuslwusl** [reply]
If you are using HA Docker you can add victoria-metrics as docker container without using this hass-addon.
Here is a docker-compose.yaml to start victoria-metrics - hope this will help you:

Just create a folder where you place the following file.
Create a folder inside called data where your long term data will be stored.
Then run ´docker-compose up´ and you should be able to use victoria-metrics.

docker-compose.yaml:
```yaml
version: '3'
services:

  victoriametrics:
    image: victoriametrics/victoria-metrics:v1.81.2
    container_name: victoriametrics
    ports:
      - "8428:8428"
    volumes:
      - ./victoriametrics/data:/data
      # - ./victoriametrics/prometheus.yml:/prometheus.yml # not required when using influxdb integration
    # environment:
    #   - TZ=Europe/London
    command:
      - "--retentionPeriod=50y"
      - "--storageDataPath=/data"
```

The plain docker commands can be found here: <https://hub.docker.com/r/victoriametrics/victoria-metrics>

Then just use my config example for homeassistant an everything should work the same as with the addon.

**Reply by paulevw** [reply]
Hey, THX, 
finally i got it installed as Addon :-) 

Now i'm just wondering, does this already change the storage behavior, so enables long term storage of the sensor data (which is what i'm aiming for) ??

**Reply by fuslwusl** [reply]
Check that you have installed version 1.8.13 or later of the victoria-metrics addon - in older versions the retention setting was not used. Then set the addon configuration setting retention to 1000y if you like. There is no special option for long term storage but 1000 years means long term storage :)

**Reply by paulevw** [reply]
I do have the 1.8.13 version, all seems to be right,
here is the start log

[victoria-metrics-start-log.txt](https://github.com/fuslwusl/homeassistant-addon-victoriametrics/files/9877335/victoria-metrics-start-log.txt)

Thing is, 
the /share/victoria-metrics-data/ folder is kind of empty, i can't find any date (base) 
so i'm not sure the data is transferred, how to judge it is long term data ...

**Reply by fuslwusl** [reply]
Your log.txt says:

```
[10:08:22] INFO: Starting Victoria Metrics with retention set to 100y
[10:08:22] INFO: Starting Victoria Metrics with args set to 
2022-10-27T08:08:22.844Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2022-10-27T08:08:22.844Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -retentionPeriod="100y"
2022-10-27T08:08:22.844Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -storageDataPath="/share/victoria-metrics-data"

```

So you have retention period of 100 years and the folder for all your long term data ist /share/victoria-metrics-data/
After 100 years victoria-metrics will start to save space by removing data older than 100 years.

**Comment by Sno88e** [comment]
How do i install this addon if I'm using hassio, is it not possible to add the repo as usual?

**Reply by fuslwusl** [reply]
Just clone the repo or download the latest addon version https://github.com/fuslwusl/homeassistant-addon-victoriametrics/tags and place the folder victoria-metrics into your homeassistant/addons folder. Then install the victoria-metrics addon.
[![Open your Home Assistant instance and show the Supervisor add-on store.](https://my.home-assistant.io/badges/supervisor_store.svg)](https://my.home-assistant.io/redirect/supervisor_store/)

**Reply by fuslwusl** [reply]
Or just add repository to the HACS Add-on Store: https://github.com/fuslwusl/homeassistant-addon-victoriametrics
![Open your Home Assistant instance and show the Supervisor add-on store.](https://my.home-assistant.io/badges/supervisor_store.svg)
This way you will get a notification in Home Assistant if there is an update of VictoriaMetrics Addon available.

**Reply by fuslwusl** [reply]
## Installation and configuration

1. Add the reposity. (Quick link: [![Open your Home Assistant instance and show the Supervisor add-on store.](https://my.home-assistant.io/badges/supervisor_store.svg)](https://my.home-assistant.io/redirect/supervisor_store/) )
    * **Add the reposity** (click 3 dots on the top right of the screen). Reposity URL: *https://github.com/fuslwusl/homeassistant-addon-victoriametrics*
    * Refresh/reload your browser tab/window
2. **Install** the add-on:
    * Find, and **install** the Victoria Metrics add-on
3. Read the add-on documentation in the *Victoria Metrics* **documentation** tab
4. **Start** the *Victoria Metrics* add-on

**Comment by mccmax1395** [comment]
Hi, 
I was wondering if there's a way to create a sensor in HA that pulls data from victoriametrics.  For example I have this basic query `count_eq_over_time(switch.espresso_machine_value[1d],1)` which just returns the count of how many times the espresso machine is turned on in a day.  Since we use influxdb to configure it, I tried adding a sensor using the config below:
```
- platform: influxdb
  api_version: 1
  queries:
      - name: "Mara X Rolling 1 Day On Count"
        where: count_eq_over_time(switch.espresso_machine_value[1d],1)
        measurement: ""
```
Unfortunately, that didn't work.  Could you point me in the correct direction to be able to do this?

**Reply by mccmax1395** [reply]
I figured it out.  I ended up using a rest sensor which is configured like this
```
- platform: rest
  resource: http://localhost:8428/api/v1/query?query=count_eq_over_time(switch.espresso_machine_value[1d],1)
  name: "Mara X Rolling 1 Day On Count"
  value_template: "{{value_json['data']['result'][0]['value'][1]}}"
```

**Comment by NorbertKre** [comment]
I'm new to HA and VictoriaMetrics - yesterday installed from scratch on Raspi 4 HA and today VM.
Running an OpenDTU and having daily data in OpenDTU and HA :-)
Have read, that VictoriaMetrics should be "the best solution" for storing long time data.

My question is, how to set-up, that data from OpenDTU will find the way to VM and later on to Grafana.
Is there anywhere a cookbook available?
